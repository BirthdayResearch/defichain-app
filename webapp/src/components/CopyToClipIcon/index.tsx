import React, { useState } from 'react';
import { I18n } from 'react-redux-i18n';
import { Popover, PopoverBody } from 'reactstrap';
import CopyToClipboard from '../CopyToClipboard';

interface Props {
  value: string;
  handleCopyFunction?: () => void;
  uid: string;
  className?: string;
}

function CopyToClipIcon(props: Props) {
  const { value, handleCopyFunction, uid, className } = props;
  const [copied, changeCopied] = useState(false);
  const handleCopy = () => {
    changeCopied(true);
    if (handleCopyFunction) {
      handleCopyFunction();
    }
    setTimeout(() => {
      changeCopied(false);
    }, 600);
  };

  return (
    <>
      <span id={uid} className={className}>
        <CopyToClipboard value={value} handleCopy={handleCopy} />
      </span>
      <Popover placement='auto' isOpen={copied} target={uid} trigger='focus'>
        <PopoverBody>{I18n.t('components.copyToClipIcon.copied')}</PopoverBody>
      </Popover>
    </>
  );
}

export default CopyToClipIcon;
