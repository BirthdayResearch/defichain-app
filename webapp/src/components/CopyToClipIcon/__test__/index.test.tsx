import React from 'react';
import CopyToClipIcon from '../index';
import { shallow } from 'enzyme';

describe('CopyToClipIcon component', () => {
  it('should check for snapshot', () => {
    const props = {
      value: 'value',
      uid: 'uid',
      className: 'className',
      handleCopyFunction: () => {},
    };

    const wrapper = shallow(<CopyToClipIcon {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
