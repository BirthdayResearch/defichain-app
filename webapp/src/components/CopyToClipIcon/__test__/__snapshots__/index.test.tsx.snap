// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyToClipIcon component should check for snapshot 1`] = `
<Fragment>
  <span
    className="className"
    id="uid"
  >
    <CopyToClipboard
      handleCopy={[Function]}
      value="value"
    />
  </span>
  <Popover
    isOpen={false}
    placement="auto"
    placementPrefix="bs-popover"
    target="uid"
    trigger="focus"
  >
    <PopoverBody
      tag="div"
    >
      copied
    </PopoverBody>
  </Popover>
</Fragment>
`;
